### DISCUSSION TITLE
Setting up Docker container to access Simple-API

### DISCUSSION THREAD

**Discussion by Leethal1977** [buanet/ioBroker.docker#344] [Share your Experience / Teile deine Erfahrung]
Set up my first Docker container in order to run iobroker and connect to my Eufy security endpoints and expose to my Hubitat hub. As a newb, the guides I could find online were great - but I had one issue that I didn't seem to be able to find any guidance on, and I'm hoping my many hours of troubleshooting may help someone avoid the same.  Docker container set up was really quick and the config of iobroker the same. Added eufy security and simple-api adapters and stood up the instances with no issues - connected straight away, Green lights, thought I was good to go. 

But every time I tried to launch simple-api, would get the connection refused error: err_connection_refused. Same with all other urls other than the admin url.

I tried everything - narrowing down to an issue with the network, binding, port forwarding in the Docker Container. Most of the threads I could find stated the start up of the container should bind to 0.0.0.0 to allow services in iobroker to listen to all ports. Went down so many rabbit holes, and in the end, all that was required was so simple - and I couldn't find this in any of the guides on setting up iobroker on Docker. 

Solution: In the Docker start-up config, add an additional port rule for port 8087 (simple-api) and one for 8082 as well (viz - if you want to use). Basically, need to add a port rule at container start up for any port in iobroker that you need to be exposed to your local host. 

Probably a no-brainer for any of you with Docker experience but I had no idea - hope it helps someone!

**Comment by buanet** [comment]
Thank you for sharing your experience!

Regards,
André
